app/app/src/main/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenter.java

```diff
@@ -15,40 +19,81 @@
     public CalculatorPresenter(CalculatorModel model, CalculatorView view) {
         this.model = model;
         this.view = view;
+        view.handleOperations(false);
     }
 
     public void onClearButtonPress() {
         model.clear();
         view.showOperationPressed(NUMBER_ZERO);
-        view.showResult(ZERO_INT);
+        view.showResult(NUMBER_ZERO);
+        view.handleDot(true);
+        view.handleOperations(false);
     }
 
-    public void onOperatorPressed(String symbol){
+    public void onOperatorPressed(String symbol) {
         model.setOperator(symbol);
         view.showOperationPressed(symbol);
+        view.handleDot(true);
     }
 
-    public void onEqualsButtonPressed(String symbol) {
-        // will be added when implements the equals button
+    public void onEqualsButtonPressed() {
+        if (!(model.getFirstNumber().isEmpty()) && !(model.getSecondNumber().isEmpty())) {
```

Comment:
in other part of code you use !model.getFirstNumber().isEmpty() and in other you use !(model.getFirstNumber().isEmpty()) pls try to use the same in all the code

Reply:
didn't see it, thanks